frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -356,17 +357,57 @@ private void downloadLicenseInfo(ResourceRequest request, ResourceResponse respo
             final LicenseInfoService.Iface licenseInfoClient = thriftClients.makeLicenseInfoClient();
             final User user = UserCacheHolder.getUserFromRequest(request);
             Project project = thriftClients.makeProjectClient().getProjectById(projectId, user);
+            excludeReleasesByChoosenRelationship(request, releaseIdsToSelectedAttachmentIds);
             LicenseInfoFile licenseInfoFile = licenseInfoClient.getLicenseInfoFile(project, user, outputGenerator,
                     releaseIdsToSelectedAttachmentIds, excludedLicensesPerAttachmentId, externalIds);
             saveLicenseInfoAttachmentUsages(project, user, selectedAttachmentIdsWithPath,
-                    excludedLicensesPerAttachmentIdWithPath);
+                    excludedLicensesPerAttachmentIdWithPath, getSelectedReleaseRationships(request));
             sendLicenseInfoResponse(request, response, project, licenseInfoFile);
         } catch (TException e) {
             log.error("Error getting LicenseInfo file for project with id " + projectId + " and generator " + outputGenerator, e);
             response.setProperty(ResourceResponse.HTTP_STATUS_CODE, Integer.toString(HttpServletResponse.SC_INTERNAL_SERVER_ERROR));
         }
     }
 
+    private void excludeReleasesByChoosenRelationship(ResourceRequest request,
+            final Map<String, Set<String>> releaseIdsToSelectedAttachmentIds) {
+
+        final List<String> relationshipsToBeIncludedList = getSelectedReleaseRationships(request);
+        Set<String> currentReleaseIdRelationships = Sets
+                .newHashSet(request.getParameterValues(PortalConstants.PROJECT_RELEASE_RELATIONS));
+
+        Map<String, String> currentReleaseIdToRelationship = extractCurrentReleaseIdToRelationship(
+                currentReleaseIdRelationships);
+
+        currentReleaseIdToRelationship = currentReleaseIdToRelationship.entrySet().stream()
+                .filter(x -> relationshipsToBeIncludedList.contains(x.getValue()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (!relationshipsToBeIncludedList.isEmpty()
+                && !relationshipsToBeIncludedList.get(0).equals(StringUtils.EMPTY)) {
+            releaseIdsToSelectedAttachmentIds.keySet().retainAll(currentReleaseIdToRelationship.keySet());
+
```

Comment:
Please remove extra line break from beginning and end of the method

Reply:
Done